citest/json_predicate/binary_predicate.py

```diff
@@ -146,6 +150,83 @@ def __call__(self, context, value):
                            path_value=PathValue('', value), valid=valid)
 
 
+class DictMatchesPredicate(BinaryPredicate):
+  """Implements binary predicate comparison predicates against dict values.
+
+  The dictionary value is in the form {field : pred} where each element
+  in the dictionary is a predicate for validating that particular field.
+  A strict predicate means that exactly all specified fields must be present.
+  Otherwise this permits additional fields with arbitrary values.
+  """
+
+  @property
+  def strict(self):
+    """Whether only the specified fields may be present (True) or not."""
+    return self.__strict
+
+  def __init__(self, operand, **kwargs):
+    """Constructor."""
+    if not isinstance(operand, dict):
+      raise TypeError(
+          '{0} is not a dict: {1!r}'.format(operand.__class__, operand))
+    self.__strict = kwargs.pop('strict', False)
+    super(DictMatchesPredicate, self).__init__('Matches', operand, **kwargs)
+
+  def __call__(self, context, value):
+    """Implements Predicate interface.
+
+    context: [ExecutionContext] The execution context.
+    value: [dict] The dictionary to match against our operand.
+
+    Returns a CompositePredicateResult indicating each of the fields.
+    """
+    if not isinstance(value, dict):
+      return TypeMismatchError(dict, value.__class__, value)
+
+    match_result_builder = predicate.CompositePredicateResultBuilder(self)
+    valid = True
+    # pylint: disable=redefined-variable-type
+    for name, pred in self.operand.items():
+      name_value = value.get(name)
+      name_result = None
+      if name_value is None:
+        name_result = MissingPathError(value, name)
+      else:
+        path_result_builder = PathPredicateResultBuilder(
+            source=value, pred=pred)
```

Comment:
Is this meant for logging/diff purposes? Seems a bit weird to store the whole dictionary for a predicate picked out by name.